server/telemetry/telemetry_client.go

```diff
@@ -203,13 +204,12 @@ func getHostname() string {
 
 func getFeatures(env environment.Env) *telpb.TelemetryFeature {
 	cache := env.GetCache()
-	authenticator := env.GetAuthenticator()
 	api := env.GetAPIService()
-
+	_, isNullAuth := env.GetAuthenticator().(*nullauth.NullAuthenticator)
```

Comment:
Nit: instead of casting here, we could add `Enabled()` to the `Authenticator` interface, which would remove the direct dependency on `nullauth` and make this kind of check less awkward to do in general.

Reply:
I would slightly prefer to leave as-is to avoid expanding the API surface of the authenticator, and also because a proper API method might encourage this type of checking.